src/main/ruby/truffleruby/core/io.rb

```diff
@@ -857,6 +857,7 @@ def initialize(fd, mode = nil, **options)
 
     @autoclose = autoclose_tmp
     @pipe = false
+    @path = path unless @path
```

Comment:
**suggestion:** Here we ignore `path` option if path is already set. I would prefer to keep this logic (I mean "ignoring") in ancestor classes.

For instance for `File` we could just reorder `super` call and assigning the `@path` variable:

```ruby
  def initialize(path_or_fd, mode = nil, perm = nil, **options)
    # ...
      super(fd, mode, **options)
      @path = path
    # ...
  end
```

Reply:
Could you elaborate on that a bit ? Do you mean that IO should still keep the "ignoring path unless set" and the descendent class File should then just set it itself ? 

I rather prefer @eregon suggestion, it's more elegant for File to just use the API of its parent like everybody else by overriding path in the **options dictionary, leaving the assignment to be done by IO.

Reply:
I also like @eregon's suggestion.